sdks/python/apache_beam/io/aws/s3io.py

```diff
@@ -464,6 +459,49 @@ def exists(self, path):
         # We re-raise all other exceptions
         raise
 
+  def _status(self, path):
```

Comment:
I see this function is only used in unit tests, and it's private. What's the purpose for it?

Reply:
This function is also used by FileSystem.metadata implementations, in order to obtain both file size and last updated fields at once (one request in web filesystems). If the file size and last_updated is collected separately it would double the api request for filesystem._list (used by match) and regression may be introduced.